sherpa/astro/ui/utils.py

```diff
@@ -6894,7 +7080,80 @@ def set_bkg(self, id, bkg=None, bkg_id=None):
         _check_type(bkg, DataPHA, 'bkg', 'a PHA data set')
         data.set_background(bkg, bkg_id)
 
-    def list_bkg_ids(self, id=None):
+    # DOC-TODO: this does not delete the source expression for the;
+    # background - is this intended or a bug?
+    def delete_bkg(self,
+                   id: Optional[IdType] = None,
+                   bkg_id: Optional[IdType] = None
+                   ) -> None:
+        """Remove the background component.
+
+        The background data set, and any associated structures - such
+        as the ARF and RMF for PHA data sets - are removed.
+
+        .. versionadded:: 4.17.0
+
+        Parameters
+        ----------
+        id : int, str, or None, optional
+           The data set to delete. If not given then the default
+           identifier is used, as returned by `get_default_id`.
+        bkg_id : int, str, or None, optional
+           The identifier for this background, which is needed if
+           there are multiple background estimates for the source.
+
+        See Also
+        --------
+        clean, delete_bkg_model, list_bkg_ids, set_bkg
+
+        Notes
+        -----
+        The source expression for the background is not removed by
+        this function.
+
+        The routine does nothing if the given dataset or background
```

Comment:
I would prefer to error out in both cases.

Reply:
I have a complex set of feelings around this

- I do appreciate the "error out" approach
- right up until I process data and write code where I have to explicitly catch the failure (or manually add validation checks) just because for this case the condition wasn't met. An example is calling "subtract" - sometimes it makes sense to error out and say "but you are already subtracted", but in other cases it can be okay to say "oh, you're already subtracted, I don't need to do anything".

I claim there's likely a range of approaches and it depends in part on whether (for Sherpa) it is a class (or low-level routine) or whether it's in the UI layer.
 
These are all generic points, and we need to think about the specifics of this. In this case I am just trying to follow the existing `delete_data` behaviour (kind of, as there are failure cases in the background case we don't have for the source case).

Reply:
I believe I was the one who insisted the `subtract` should not give me an error if it's already subtracted because I run into cases where some of my datasets were already subtracted and some are not and then it failed in the middle (#766). So, point taken.